### PR TITLE
Helm from GitHub

### PR DESCRIPTION
**Pull request by rickducott**
BOT NOTES: 
resolves https://github.com/solo-io/go-utils/issues/121
resolves https://github.com/solo-io/go-utils/issues/122

### CODE REVIEW

File: fsutils/afero.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
+package fsutils
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/spf13/afero"
+)
+
+func SetupTemporaryFiles(fs afero.Fs) (file afero.File, dir string, err error) {
```
Comment (mitchdraft) [inline fsutils/afero.go]: I think a utils lib like this is useful, but I feel this function is too specific to get any reuse outside of github archive downloads. The resources it creates `tar-dir-` and `tar-file-` are pretty specific.
How about moving this to the helm chart package where it's used or renaming it something like `SetupTemporaryFilesForGitArchive`

File: fsutils/afero.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	return tmpf, tmpd, err
+}
+
+func GetRepoFolder(fs afero.Fs, tmpd string) (string, error) {
+	files, err := afero.ReadDir(fs, tmpd)
+	if err != nil {
+		return "", err
+	}
+	if len(files) != 1 {
+		return "", fmt.Errorf("expected only one folder from archive tar, found (%d)", len(files))
```
Comment (mitchdraft) [inline fsutils/afero.go]: similar comment to above, this util has some strong assumptions, consider privatizing it or renaming

File: installutils/helmchart/manifests_test.go [hunk 3]
```diff
@@ -91,4 +91,30 @@ metadata:
 
 		})
 	})
+
+	Context("load from github", func() {
+		It("works for a single chart", func() {
+			ref := GithubChartRef{
+				Ref:            "c00d740c83ec5b5439bf2f58fee110a546af7177",
```
Comment (mitchdraft) [inline installutils/helmchart/manifests_test.go]: good idea locking to a ref, was going to suggest you make some dummy helm resources in this repo's test resources directory. Might still be worth doing so you can test various helm ignore rules.
Though it produces a chicken-and-egg problem, can't test against it till it merges

File: fsutils/afero.go [hunk 4]
```diff
@@ -0,0 +1,42 @@
+package fsutils
```
Comment (mitchdraft) [inline fsutils/afero.go]: consider calling this `vfsutils` as it seems to be oriented towards virtual filesystems

File: fsutils/afero.go (hunk 1 above)
Comment (rickducott) [reply fsutils/afero.go]: So I actually copied them over from solobot and didn’t want to modify or make private here so I could delete them there. Don’t mind leaving them in helm / install utils, but in the solobot case it’s for changelog verification, nothing to do with helm.

File: installutils/helmchart/manifests_test.go (hunk 3 above)
Comment (rickducott) [reply installutils/helmchart/manifests_test.go]: FWIW the helm ignore rules are designed to be used during packaging, not rendering. So I'm not sure there's a ton of value in adding testing around that. That said, I did take a look at a few of the charts and rendered manifests just to make sure things checked out.

File: fsutils/afero.go (hunk 1 above)
Comment (rickducott) [reply fsutils/afero.go]: On second thought, you're right -- added a better utility and moved these to private helpers. Should be pretty easy to make use of the new function in solobot.

File: fsutils/afero.go (hunk 2 above)
Comment (rickducott) [reply fsutils/afero.go]: fixed

**Comment by rickducott** [comment]
Thanks @mitchdraft, done iterating -- could you take another look?

**Pull request merged**